Mathlib/Data/Polynomial/Coeff.lean

```diff
@@ -293,16 +293,16 @@ theorem mul_X_pow_eq_zero {p : R[X]} {n : ℕ} (H : p * X ^ n = 0) : p = 0 :=
   ext fun k => (coeff_mul_X_pow p n k).symm.trans <| ext_iff.1 H (k + n)
 #align polynomial.mul_X_pow_eq_zero Polynomial.mul_X_pow_eq_zero
 
-theorem mul_X_pow_injective (n : ℕ) : Function.Injective fun P : R[X] => X ^ n * P := by
+-- TODO Unify this with `Polynomial.Monic.isRegular`
+theorem isRegular_X_pow (n : ℕ) : IsRegular ((X : R[X]) ^ n) := by
```

Comment:
nit (untested):
```suggestion
theorem isRegular_X_pow (n : ℕ) : IsRegular (X ^ n : R[X]) := by
```

Reply:
While we are on the topic, @eric-wieser, what do you think of `(X (R := R) ^ n)`?

Reply:
I think that's worse

Reply:
I agree with Eric here and have made this suggestion.